### PR TITLE
Use selected theme to set default highlight color setting value

### PR DESCRIPTION
**Pull request by dalthviz**
# References and relevant issues

Follow up of #6689 

# Description

Get a default value for the highlight color setting from the current selected theme. The setting is set to the theme corresponding value only if the user has not changed the setting before.

### Theme `current` color

* light theme color: `rgb(253, 240, 148)`
* dark theme color: `rgb(0, 122, 204)`

A preview:

theme_color_default_highlight

### New `border_highlight` color from the themes

* light theme color: `rgb(206, 196, 121)`
* dark theme color: `rgb(135, 206, 235)`

A preview;

theme_color_default_highlight_2

# Notes:

Couple of things I'm not completely sure about so I left them as `TODOs` for the moment and so any feedback related to them is appreciated!:

* [x] ~The themes have a `highlight` value but is a kind of dark color (dark theme: `rgb(106, 115, 128)`, light theme: `rgb(163, 158, 156)`). Checking, seems like a more suitable color from the current values available in the themes is the `current` color (dark theme: `rgb(0, 122, 204)`, light theme: `rgb(253, 240, 148)`). Should the `current` color from the themes be used or should a new value be added something like `data_highlight`? If a new value is added to the themes what value it should have for dark and light themes?~
         * Go with a new `border_highlight` color value to set more suitable colors (light theme `current` color doesn't have much contrast/looks quite invisible)
* [x]  ~This PR as it is only covers using the theme to get a color, should also a value from the theme reflect the default highlight thickness value?~
         * Use the theme just to get a color, get more feedback regarding the theme usage
* [ ] To enable the highlight widgets to get the theme color when the theme changes I had to add some events connections in the `AppearanceSettings` (adding a `_highlight_changed` method connected to the `highlight.events`). Not totally sure if there is a better approach for that or maybe is not necessay and I'm missing something besides the changes done in the `AppearanceSettings.update` and `AppearanceSettings.__setattr__` methods to handle the `highlight.highlight_color` value when the `theme` being selected changes 🤔

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
> The themes have a highlight value but is a kind of dark color (dark theme: rgb(106, 115, 128), light theme: rgb(163, 158, 156)). Checking, seems like a more suitable color from the current values available in the themes is the current color (dark theme: rgb(0, 122, 204), light theme: rgb(253, 240, 148)). Should the current color from the themes be used or should a new value be added something like data_highlight? If a new value is added to the themes what value it should have for dark and light themes?

Pretty sure that `highlight` is for buttons and icons, etc. So IMO we should leave it and make a `border_highlight`?

**Comment by brisvag** [comment]
I kinda like using the "current" color, it would match the meaning in the GUI sense. On the other hand, that yellow looks invisible, so I'm not sure where actually that's getting used in the gui...

Ultimately, not strong feelings for me here :shrug: 

> This PR as it is only covers using the theme to get a color, should also a value from the theme reflect the default highlight thickness value?

This seems wrong to me, it feels "outside" the scope of the theme. Buyt maybe it's a hint that the whole thing shouldn't be part of the theme? ^^' sorry, conflicting thoughts :P Let's see what others have to say.

**Comment by dalthviz** [comment]
Thanks for the feedback! So I would say:

> Pretty sure that highlight is for buttons and icons, etc. So IMO we should leave it and make a border_highlight for the new functionality?

That makes sense :+1:

> I kinda like using the "current" color, it would match the meaning in the GUI sense. On the other hand, that yellow looks invisible, so I'm not sure where actually that's getting used in the gui...
> 
> Ultimately, not strong feelings for me here 🤷

I think the style files use the theme `current` color but with some processing done via the `darken` and `lighten` functions (which most probably makes things look better 🤔)

> This seems wrong to me, it feels "outside" the scope of the theme. Buyt maybe it's a hint that the whole thing shouldn't be part of the theme? ^^' sorry, conflicting thoughts :P Let's see what others have to say.

Yep, having the thickness seems pretty strange to me too 😅 I guess the color value at least is something more related with what the theme provides (colors that the interface uses) so kind of makes sense 🤔 but yep happy to hear what others think too :+1:

In the main time, for the light theme I think the color used for things with `darken(current)` is something like `rgb(206, 196, 121)` so I will use that in a new `border_highlight` color. For the dark theme, I was thinking in using the previous default highlight color, so I will set the `skyblue` color (`rgb(135, 206, 235)`) as `border_highlight` there.

Let me know if you have any further feedback and thanks again for giving this a check!

**Comment by psobolewskiPhD** [comment]
> `rgb(206, 196, 121)` so I will use that in a new `border_highlight` color.

Bleh! I kinda hate the way that looks. I'd prefer a slightly darker skyblue (same for the Dark theme, the skyblue is too faint IMO). 
(If I had my druthers, in light theme the highlight in the layer list and buttons would also be blue, rather than the yellow, but that's a different issue...)
As I said in the issue, the highlight is "close" to the data being visualized and versus which it needs to provide contrast, which is why I think it should have some sane defaults but be user-editable. It's less important it coordinate with the rest of the viewer window -- as opposed to things like widgets and buttons. So in the theme or not, for me isn't a big deal.

I think if color is in the theme, then so should the thickness, it's just more intuitive I think given the widget presentation.
(I personally think 1 as default is too thin because it's 1 screen pixel so it's super faint on my displays and flat out disappears when zoomed out a bit -- maybe this could consider the DPI scaling? Different PR I guess--I'll keep resetting it to 2 or 3.)

I will say I think the behavior with switching themes is strange:
When I open the preferences in fresh install (default Dark) I have sky-blue in the widget and it highlights as that.
If I switch to Light it shows sky blue in the widget -- not updated? But the actual highlights will be the new sandy color. So it's a bit unclear what's going on.

File: napari/settings/_appearance.py [hunk 1]
```diff
@@ -60,22 +76,36 @@ def update(
...
+            current_highlight_color = parse_color_as_float_list(
+                current_theme.border_highlight
+            ) + [1.0]
+            new_highlight_color = parse_color_as_float_list(
+                new_theme.border_highlight
+            ) + [1.0]
+
             if values['font_size'] == int(current_theme.font_size[:-2]):
                 values['font_size'] = int(new_theme.font_size[:-2])
+            if np.array_equal(
```
Comment (psobolewskiPhD) [inline napari/settings/_appearance.py]:
```suggestion
            if np.allclose(
```
I think float precision is why changing theme doesn't change the widget color.

**Comment by dalthviz** [comment]
> I rather dislike the way that looks on shapes with the default grey shape edge. It looks better with other shape colors... I'd prefer blue, but this sandy color is consistent with the theme....
(If I had my druthers, in light theme the highlight in the layer list and buttons would also be blue, rather than the yellow, but that's a different issue...)

Maybe an issue regarding a revision for the theme colors could be worthy (I think the yellow and white color combination is quite low contrast so indeed it could be improved towards a better color combination)

> As I said in the issue, the highlight is "close" to the data being visualized and versus which it needs to provide contrast, which is why I think it should have some sane defaults but be user-editable. It's much less important it coordinate with the rest of the viewer window (aka the theme) -- as opposed to things like widgets and buttons. So in the theme or not, for me isn't a big deal.

Maybe for the moment is better to just choose a good default color for the highlight that works with the light and dark theme instead of trying to connect it directly with the theme?

> I think if color is in the theme, then so should the thickness, it's just more intuitive I think given the widget presentation.
(I personally think 1 as default is too thin because it's 1 screen pixel so it's super faint on my displays and flat out disappears when zoomed out a bit -- maybe this could consider the DPI scaling? Different PR I guess--I'll keep resetting it to 2 or 3.)

Seems to me that connecting the highlight to the theme is not the best then. Maybe actually was is needed instead is to define better default values for the highlight setting (not just color but also thickness)?

> I will say I think the behavior with switching themes is strange:
When I open the preferences in fresh install (default Dark) I have sky-blue in the widget and it highlights as that.
If I switch to Light it shows sky blue in the widget -- not updated? But the actual highlights will be the new sandy color. So it's a bit unclear what's going on.

I was able to reproduce this and indeed that's not the intended behavior with the changes here. The idea would be for the widget previewing the highlight color to show the theme defined color (dark theme -> skyblue; light theme -> sandy color ) if the user has not manually changed the selected color and the current selected theme is changed. I accepted the suggestion to change `array_equal` for `allclose` and that worked for one run but then I got again the widget not updating when I checked what would happen with clean settings (running `napari --reset`).  Seems like something else is needed to trigger a widget update for the highlight color after the theme setting changes 🤔

**Comment by brisvag** [comment]
Thanks for exploring this @dalthviz! But yup, after seeing the above conversion, *color* me convinced :wink:

**Comment by dalthviz** [comment]
Created #6781 so closing this. Thanks for the time checking this and all the feedback @psobolewskiPhD @brisvag !

**Pull request closed without merging**